### PR TITLE
Start switching to use slog

### PR DESCRIPTION
**Pull request by norkans7**

### CODE REVIEW

File: handlers/jiochat/handler_test.go [hunk 1]
```diff
@@ -20,6 +20,7 @@ import (
 	"github.com/nyaruka/gocommon/urns"
 	"github.com/sirupsen/logrus"
 	"github.com/stretchr/testify/assert"
+	"golang.org/x/exp/slog"
```
Comment (rowanseymour) [inline handlers/jiochat/handler_test.go:23]: "log/slog"

File: handlers/wechat/handler_test.go [hunk 2]
```diff
@@ -20,6 +20,7 @@ import (
 	"github.com/nyaruka/gocommon/urns"
 	"github.com/sirupsen/logrus"
 	"github.com/stretchr/testify/assert"
+	"golang.org/x/exp/slog"
```
Comment (rowanseymour) [inline handlers/wechat/handler_test.go:23]: "log/slog"

File: queue/queue.go [hunk 3]
```diff
@@ -6,7 +6,7 @@ import (
 	"time"
 
 	"github.com/gomodule/redigo/redis"
-	"github.com/sirupsen/logrus"
+	"golang.org/x/exp/slog"
```
Comment (rowanseymour) [inline queue/queue.go:9]: "log/slog"

File: server.go [hunk 4]
```diff
@@ -22,6 +22,7 @@ import (
 	"github.com/nyaruka/gocommon/jsonx"
 	"github.com/pkg/errors"
 	"github.com/sirupsen/logrus"
+	"golang.org/x/exp/slog"
```
Comment (rowanseymour) [inline server.go:25]: "log/slog"

**Review by rowanseymour — CHANGES_REQUESTED** [review]

File: cmd/courier/main.go [hunk 5]
```diff
@@ -1,13 +1,17 @@
 package main
 
 import (
+	ulog "log"
+	"log/slog"
 	"os"
 	"os/signal"
+	goruntime "runtime"
```
Comment (rowanseymour) [inline cmd/courier/main.go:8]: this is only necessary in mailroom because we have our own `runtime` struct

Comment (norkans7) [reply cmd/courier/main.go]: reverted that

File: server.go (hunk 4 above)
Comment (norkans7) [reply server.go]: replaced properly

Comment (rowanseymour) [reply server.go]: the `golang.org/x/exp` is where new experimental features are tried out before sometimes becoming part of the golang std lib - so `slog` first existed there before being added to go 1.21

File: handlers/jiochat/handler_test.go [hunk 6]
```diff
@@ -13,6 +13,8 @@ import (
 	"testing"
 	"time"
 
+	"log/slog"
```
Comment (rowanseymour) [inline handlers/jiochat/handler_test.go:16]: no space needed between this and other deps - standard go formatting is only a space between std lib deps and other deps

File: go.sum [hunk 7]
```diff
@@ -25,6 +25,7 @@ github.com/getsentry/raven-go v0.2.0/go.mod h1:KungGk8q33+aIAZUIVWZDr2OfAEBsO49P
 github.com/go-chi/chi v4.1.2+incompatible h1:fGFk2Gmi/YKXk0OmGfBh0WgmN3XB8lVnEyNz34tQRec=
 github.com/go-chi/chi v4.1.2+incompatible/go.mod h1:eB3wogJHnLi3x/kFX2A+IbTBlXxmMeXJVKy9tTv1XzQ=
 github.com/go-playground/assert/v2 v2.2.0 h1:JvknZsQTYeFEAhQwI4qEt9cyV5ONwRHC+lYKSsYSR8s=
+github.com/go-playground/assert/v2 v2.2.0/go.mod h1:VDjEfimB/XKnb+ZQfWdccd7VUvScMdVu0Titje2rxJ4=
```
Comment (rowanseymour) [inline go.sum:28]: go mod tidy

Comment (norkans7) [reply go.sum:28]: yes

File: queue/queue.go [hunk 8]
```diff
@@ -201,7 +201,7 @@ func PopFromQueue(conn redis.Conn, qType string) (WorkerToken, string, error) {
 	epochMS := strconv.FormatFloat(float64(time.Now().UnixNano()/int64(time.Microsecond))/float64(1000000), 'f', 6, 64)
 	values, err := redis.Strings(luaPop.Do(conn, epochMS, qType))
 	if err != nil {
-		logrus.Error(err)
+		slog.Error("error poping from queue", "error", err)
```
Comment (rowanseymour) [inline queue/queue.go:204]: popping

**Review by rowanseymour — APPROVED** [review]

File: queue/queue.go (hunk 8 above)
Comment (norkans7) [reply queue/queue.go]: fixed

**Review by rowanseymour — APPROVED** [review]

**Pull request merged**